applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -108,6 +122,68 @@ static struct bt_audio_discover_params audio_source_discover_param[CONFIG_BT_MAX
 static int discover_source(struct bt_conn *conn);
 #endif /* CONFIG_STREAM_BIDIRECTIONAL */
 
+/**
+ * @brief  Get the presentation delay for all headsets
+ *
+ * @param  mode   The mode of search for the presentation delay
+ * @param  index  The index of the headset to test against
+ *
+ * @return The chosen presentation delay in micro seconds or 0 if error
+ */
+static uint32_t find_headset_pres_delay(enum le_audio_pres_search mode, uint8_t index)
+{
+	uint32_t pres_dly_us = 0;
+	uint32_t pres_dly_min = headsets[index].sink_ep->qos_pref.pd_min;
+	uint32_t pres_dly_max = headsets[index].sink_ep->qos_pref.pd_max;
+	uint32_t pref_dly_min = headsets[index].sink_ep->qos_pref.pref_pd_min;
+	uint32_t pref_dly_max = headsets[index].sink_ep->qos_pref.pref_pd_max;
+
+	for (int i = 0; i < ARRAY_SIZE(headsets); i++) {
+		if (headsets[i].sink_ep != NULL) {
+			pres_dly_min = MAX(pres_dly_min, headsets[i].sink_ep->qos_pref.pd_min);
+			pres_dly_max = MIN(pres_dly_max, headsets[i].sink_ep->qos_pref.pd_max);
+			pref_dly_min = MAX(pref_dly_min, headsets[i].sink_ep->qos_pref.pref_pd_min);
+			pref_dly_max = MIN(pref_dly_max, headsets[i].sink_ep->qos_pref.pref_pd_max);
+		}
+	}
+
+	switch (mode) {
+	case AUDIO_PRES_DELAY_MODE_MIN:
+		pres_dly_us = pres_dly_min;
+		break;
+
+	case AUDIO_PRES_DELAY_MODE_MAX:
+		pres_dly_us = pres_dly_max;
+		break;
+
+	case AUDIO_PRES_DELAY_MODE_PREF_MIN:
+		pres_dly_us = pref_dly_min;
+		break;
+
+	case AUDIO_PRES_DELAY_MODE_PREF_MAX:
+		pres_dly_us = pref_dly_max;
+		break;
+
+	case AUDIO_PRES_DELAY_MODE_LOCAL:
+		if (IN_RANGE(CONFIG_BT_AUDIO_LOCAL_PRESENTATION_DELAY_US, pres_dly_min,
+			     pres_dly_max)) {
+			pres_dly_us = CONFIG_BT_AUDIO_LOCAL_PRESENTATION_DELAY_US;
+		} else {
+			pres_dly_us = 0;
+			LOG_DBG("Preferred local presentaion delay outside of range");
```

Comment:
```suggestion
			LOG_WRN("Preferred local presentaion delay outside of range");
```

Reply:
Actually this probably should be LOG_ERR and return -EINVAL.

Reply:
I disagree. If the local is outside the range, but the pres_dly_min or pres_dly_max is within the range of the gateway then the gateway should follow suit as opposed to returning an error and not having a common presentation delay.

Reply:
This has the potential to be a poor user experience and hence any products would be poorly reviewed. Wouldn't it be better to not allow the connection and pass the decision back to the customer. For example if lip sync is off this would be very annoying to the user. 

All that said, we could take the decision to not error and use the the closest delay to this local out of range delay. However we would need to change the search and document this as a customer decision.

Could we discuss this?

Reply:
Updated so that the max or min delay is returned if the local delay is out of range, depending on if it is greater or less than.